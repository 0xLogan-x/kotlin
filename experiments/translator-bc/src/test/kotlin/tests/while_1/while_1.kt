fun while_test_1(x: Int): Int {
    var a = 5
    while (2 < 3) {
        a = 10
        return a
    }
    return 5
}