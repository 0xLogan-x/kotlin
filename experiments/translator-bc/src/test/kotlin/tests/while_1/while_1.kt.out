@exceptions.str.1 = private unnamed_addr constant  [59 x i8] c"Exception in thread main kotlin.KotlinNullPointerException\00", align 1
declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i32, i1)
declare i8* @malloc_heap(i32)
declare i32 @printf(i8*, ...)
declare void @abort()
%class.Nothing = type { }
define weak i32 @while_test_1_Int(i32  %while_test_1.x) #0
{
%while_test_1.x.addr = alloca i32, align 4
store i32 %while_test_1.x, i32* %while_test_1.x.addr, align 4
%managed.unique.0.while_test_1.a = alloca i32, align 4
%var2 = load i32* %managed.unique.0.while_test_1.a, align 4
store i32 5, i32* %managed.unique.0.while_test_1.a, align 4
br label %label.while.1
label.while.1:
%var3 = icmp slt i32 2, 3
br i1 %var3, label %label.while.2, label %label.while.3
label.while.2:
%var4 = load i32* %managed.unique.0.while_test_1.a, align 4
store i32 10, i32* %managed.unique.0.while_test_1.a, align 4
%var5 = load i32* %managed.unique.0.while_test_1.a, align 4
ret i32 %var5
br label %label.while.1
label.while.3:
ret i32 5
unreachable
}

